activerecord/test/cases/relation/where_test.rb

```diff
@@ -316,6 +316,31 @@ def test_where_with_blank_conditions
       end
     end
 
+    def test_where_with_integer_for_string_column_using_prepared_statement
+      count = Post.where("title = ?", 0).count
+      assert_equal 0, count
+    end
+
+    def test_where_with_float_for_string_column_using_prepared_statement
+      count = Post.where("title = ?", 0.0).count
+      assert_equal 0, count
+    end
+
+    def test_where_with_boolean_for_string_column_using_prepared_statement
+      count = Post.where("title = ?", false).count
+      assert_equal 0, count
+    end
+
+    def test_where_with_decimal_for_string_column_using_prepared_statement
+      count = Post.where("title = ?", BigDecimal(0)).count
+      assert_equal 0, count
+    end
+
+    def test_where_with_duration_for_string_column_using_prepared_statement
+      count = Post.where("title = ?", 0.seconds).count
+      assert_equal 0, count
```

Comment:
This test fails and returns `11` without the above change. All the other tests pass as expected without change.